src/sinks/util/buffer/metrics.rs

```diff
@@ -350,18 +366,21 @@ mod test {
             .collect()
     }
 
-    fn rebuffer<State: MetricNormalize>(events: Vec<Metric>) -> Buffer {
+    fn rebuffer<State: MetricNormalize>(metrics: Vec<Metric>) -> Buffer {
         let batch_size = BatchSettings::default().bytes(9999).events(6).size;
-        let mut buffer = MetricsBuffer::<State>::new(batch_size);
+        let mut normalizer = MetricNormalizer::<State>::default();
+        let mut buffer = MetricsBuffer::new(batch_size);
         let mut result = vec![];
 
-        for event in events {
-            match buffer.push(Event::Metric(event)) {
-                PushResult::Overflow(_) => panic!("overflowed too early"),
-                PushResult::Ok(true) => {
-                    result.push(buffer.fresh_replace().finish());
+        for metric in metrics {
+            if let Some(event) = normalizer.apply(Event::Metric(metric)) {
```

Comment:
Under which condition does this evaluate to `None`, and is no handling needed in that case?

Reply:
The normalizer emits `None` if it absorbs the metric into its state. For example, when converting absolute metrics to incremental, the first metric in a series is used to set the state and so would return `None` here, and then each subsequent one would result in `Some(…)`.

Reply:
Thanks! I think we should add the explanation for the `Option` return type in a comment to the `MetricNormalizer::apply` and `MetricNormalize::apply_state` methods as well.

A follow up question, but I'm not sure if this applies to the same context of your answer: In the case of absolute -> incremental, wouldn't the first increment correspond to the absolute value?

Reply:
Yes, better internal documentation is always a good idea. I'll add some.

The first increment corresponding to the absolute value only works if the previous value is assumed to be zero. That could introduce big spikes in the time series if it is not, though.

Reply:
Does that mean that the metric provider should emit a `0` first if the first value indeed was supposed to be absolute?

I don't know too much about the context in which this buffer is used, this edge case just came up thinking about it.

Reply:
Just discussed with Bruce: This can indeed cause some janky behavior. If the first seed value would be emitted as increment, the metric would have some big jumps when Vector restarts. On the other hand, if it is simply ignored, values like "uptime seconds" could show inaccurate/misleading values when the seed value is not emitted.

Any solution to this is vastly out of scope for this PR, though.